lib/security.c

```diff
@@ -406,13 +402,11 @@ int Curl_sec_read_msg(struct connectdata *conn, char *buffer,
 
   if(buf[decoded_len - 1] == '\n')
     buf[decoded_len - 1] = '\0';
-  /* FIXME: Is |buffer| length always greater than |decoded_len|? */
```

Comment:
This one looks like it could be significant too. The FIXME comment was added in 7d4f8c2 "Curl_sec_read_msg tweaks" but the logic has been there for 19 years https://github.com/curl/curl/commit/def69c30879c0246bccb02d79e06b937e39d0ba4#diff-cfb4f92cb398620c174d520bc6aae71eR359. I can't find a discussion reference. @jchaffraix @bagder

Reply:
I don't think there ever was a discussion about that detail.